lib/api/v3/work_packages/work_package_contract.rb

```diff
@@ -27,30 +27,57 @@
 # See doc/COPYRIGHT.rdoc for more details.
 #++
 
-require 'reform'
-require 'reform/form/active_model/model_validations'
-
 module API
   module V3
     module WorkPackages
-      class WorkPackageContract < Reform::Contract
-        def self.writable_attributes
-          @writable_attributes ||= %w(
-            lock_version
-            subject
-            parent_id
-            description
-            start_date
-            due_date
-            status_id
-            assigned_to_id
-            responsible_id
-            priority_id
-            category_id
-            fixed_version_id
-            done_ratio
-            estimated_hours
-          )
+      class WorkPackageContract < ::API::Contracts::ModelContract
+        attribute :subject
+        attribute :description
+        attribute :start_date, :due_date
+        attribute :status_id
+        attribute :priority_id
+        attribute :category_id
+        attribute :fixed_version_id
+
+        attribute :lock_version do
+          errors.add :error_conflict, '' if model.lock_version.nil? || model.lock_version_changed?
+        end
+
+        attribute :parent_id do
+          if model.changed.include? 'parent_id'
+            errors.add :error_unauthorized, '' unless @can.allowed?(model, :manage_subtasks)
+          end
+        end
+
+        attribute :assigned_to_id do
+          validate_people_visible :assignee,
+                                  'assigned_to_id',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.